ecosystem/typescript/sdk/examples/typescript-esm/offer_capabilities.ts

```diff
@@ -0,0 +1,134 @@
+import { AptosAccount, FaucetClient, Network, Provider, HexString, TxnBuilderTypes, BCS, Types } from "aptos";
+import assert from "assert";
+
+const CORE_CODE_ADDRESS = new HexString("0x0000000000000000000000000000000000000000000000000000000000000001");
+const ED25519_ACCOUNT_SCHEME = 0;
+
+// Works for both SignerCapabilityOffer and RotationCapabilityOffer.
+// The structName changes and chainId isn't used in SignerCapabilityOffer
+type CapabilityOfferProofChallengeV2 = {
+  accountAddress: TxnBuilderTypes.AccountAddress;
+  moduleName: string;
+  structName: string;
+  sequenceNumber: number;
+  sourceAddress: TxnBuilderTypes.AccountAddress;
+  recipientAddress: TxnBuilderTypes.AccountAddress;
+  chainId?: number; // not used in SignerCapabilityOffer
+};
+
+const createAndFundAliceAndBob = async (
+  faucetClient: FaucetClient,
+): Promise<{ alice: AptosAccount; bob: AptosAccount }> => {
+  console.log(`---------------  Creating and funding a new Bob & Alice  ---------------`);
+  const alice = new AptosAccount();
+  const bob = new AptosAccount();
+  await faucetClient.fundAccount(alice.address(), 100_000_000);
+  await faucetClient.fundAccount(bob.address(), 100_000_000);
+  console.log({
+    alice: alice.address().toString(),
+    bob: bob.address().toString(),
+  });
+  return {
+    alice,
+    bob,
+  };
+};
+
+(async () => {
+  const provider = new Provider(Network.DEVNET);
+  const faucetClient = new FaucetClient(provider.aptosClient.nodeUrl, "https://faucet.devnet.aptoslabs.com");
+
+  // Offer rotation capability
+  {
+    // Create and fund two new accounts
+    const { alice, bob } = await createAndFundAliceAndBob(faucetClient);
+    // Note that the rotation capability offer needs the chainId
+    console.log(`---------------  RotationCapabilityOfferProofChallengeV2 --------------`);
+    const { hash, version, success, payload } = await signStructAndSubmitTransaction(
+      provider,
+      alice,
+      "offer_rotation_capability",
+      {
+        accountAddress: TxnBuilderTypes.AccountAddress.fromHex(CORE_CODE_ADDRESS),
+        moduleName: "account",
+        structName: "RotationCapabilityOfferProofChallengeV2",
+        sequenceNumber: Number((await provider.getAccount(alice.address())).sequence_number),
+        sourceAddress: TxnBuilderTypes.AccountAddress.fromHex(alice.address()),
+        recipientAddress: TxnBuilderTypes.AccountAddress.fromHex(bob.address()),
+        chainId: await provider.aptosClient.getChainId(),
```

Comment:
Generally, I'm in favor of doing these prior to the call instead of inside them for readability / sequencing.  Vs right now, the inner calls happen, then the outer.